### PR TITLE
Replace TensorCompare with TensorMemoryCompare to handle the Nan tens…

### PR DESCRIPTION
**Pull request by JackCaoG**
…or comparsion.

### CODE REVIEW

File: torch_xla/csrc/tensor_util.cpp [hunk 1]
```diff
@@ -670,17 +661,13 @@ at::Tensor MakeTensorFromXlaLiteral(const xla::Literal& literal,
   }
 }
 
-bool TensorCompare(const at::Tensor& t1, const at::Tensor& t2) {
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_util.cpp:673]: @JackCaoG Leave the API "memory less". Just call TensorMemoryCompare() and add a comment about why.
I feel this to be a temp solution, as the compare of two aten tensors should be deterministic, fast-math or not.

File: torch_xla/csrc/tensor_util.h [hunk 2]
```diff
@@ -25,6 +25,8 @@ std::vector<at::Tensor> XlaDataToTensors(
     absl::Span<const xla::ComputationClient::DataPtr> xla_data,
     at::ScalarType dest_element_type);
 
+bool TensorMemoryCompare(const at::Tensor& t1, const at::Tensor& t2);
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_util.h:28]: Do not expose here. No need.

File: torch_xla/csrc/tensor_util.cpp [hunk 3]
```diff
@@ -670,6 +661,15 @@ at::Tensor MakeTensorFromXlaLiteral(const xla::Literal& literal,
   }
 }
 
+bool TensorMemoryCompare(const at::Tensor& t1, const at::Tensor& t2) {
```
Comment (dlibenzi) [inline torch_xla/csrc/tensor_util.cpp:664]: Undo this change. Leave in anon ns.

Comment (JackCaoG) [reply torch_xla/csrc/tensor_util.cpp:664]: so we just have tensorCompare to directly call TensorMemoryCompare in all cases?

Comment (dlibenzi) [reply torch_xla/csrc/tensor_util.cpp:664]: Yes. Leave a comment there telling about non-deterministic ATEM compare in presence of NaN.
